civis/futures.py

```diff
@@ -84,6 +84,62 @@ def __init__(self, poller, poller_args,
                          client=client,
                          poll_on_creation=poll_on_creation)
 
+        self._exception_handled = False
+        self.add_done_callback(self._set_job_exception)
+
+    @staticmethod
+    def _set_job_exception(fut):
+        """Callback: On job completion, check the status.
+        If the job has failed, and has no pre-existing error message,
+        populate the error message with recent logs.
+        """
+        # Prevent infinite recursion: this function calls `set_exception`,
+        # which triggers callbacks (i.e. re-calls this function).
+        if fut._exception_handled:
+            return
+        else:
+            fut._exception_handled = True
+
+        if fut.failed():
+            # Container scripts do not return the error message,
+            # so we override with an exception we can pull from the
+            # logs
+            if isinstance(fut._exception, CivisJobFailure) and \
+               fut._exception.error_message == 'None':
```

Comment:
I'm not sure how likely it is, but maybe since this is covering a wider range of objects this check can be done on `'None'`, `None`, and `''`.
This could also be added as a property on the `CivisJobFailure` class since we know it is a job failure at this point. Something like `fut._exception.has_empty_error_message`